driver.py

```diff
@@ -10,9 +11,8 @@ class Driver(object):
     buf = None
     def __new__(cls, *args, **kwargs):
         if not cls._instance:
-            cls._instance = super(Driver, cls).__new__(
-                                cls, *args, **kwargs)
-            cls.f = open('/dev/xmas', 'w')
+            cls._instance = super(Driver, cls).__new__(cls)
```

Comment:
Does Python3 not need to pass *args, **kwargs to the super's __new__? I guess in this case object probably doesn't take those so maybe it doesn't matter.

Reply:
In Python 3 __new__ takes exactly one argument, cls, so I had to do this to make it compile.